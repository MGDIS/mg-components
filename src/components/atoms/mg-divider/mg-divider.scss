.mg-divider {
  margin: var(--mg-divider-vertical-spacing) auto;
  width: 12rem;
  height: var(--mg-divider-thickness);
  background-color: var(--mg-divider-background-color);

  &.mg-divider--full {
    width: 100%;
  }
}
