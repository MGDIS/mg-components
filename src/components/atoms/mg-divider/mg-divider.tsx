import { Component, h, Prop } from '@stencil/core';

@Component({
  tag: 'mg-divider',
  styleUrl: 'mg-divider.scss',
  shadow: true,
})
export class MgDivider {
  /**************
   * Decorators *
   **************/

  /**
   * Define component size
   */
  @Prop() size: 'regular' | 'full' = 'regular';

  /**
   * Render
   *
   * @returns {HTMLElement} HTML Element
   */
  render(): HTMLElement {
    return (
      <div
        class={{
          'mg-divider': true,
          'mg-divider--full': this.size === 'full',
        }}
      ></div>
    );
  }
}
