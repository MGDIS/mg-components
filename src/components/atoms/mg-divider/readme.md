## Sizing

![](./mg-divider/doc/img/mg-divider-sizing.png)

Two sizes are possible :

- Regular : 120px
- Full : 100%

Rules :

- Regular is the default mode.
- In full mode it takes 100% of its parent.
- The thickness is 1px.

## Spacing

![](./mg-divider/doc/img/mg-divider-spacing.png)

Default margin applied on top and bottom of the divider is set to 40px.

## Color

![](./mg-divider/doc/img/mg-divider-color.png)

## CSS Variables

If needed some [variables](./?path=/story/css-variables--page) are available to customize the component:

- `--mg-divider-background-color`: define the divider background color, default: `hsla(var(--color-danger), 15%)`
- `--mg-divider-thickness`: define the divider thickness, default: `0.1rem`
- `--mg-divider-vertical-spacing`: define the divider vertical spacing, default: `4rem`

<!-- Auto Generated Below -->


## Properties

| Property | Attribute | Description           | Type                  | Default     |
| -------- | --------- | --------------------- | --------------------- | ----------- |
| `size`   | `size`    | Define component size | `"full" \| "regular"` | `'regular'` |


----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
