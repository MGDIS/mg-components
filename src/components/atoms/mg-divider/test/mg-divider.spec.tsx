import { h } from '@stencil/core';
import { newSpecPage } from '@stencil/core/testing';
import { MgDivider } from '../mg-divider';

const getPage = args =>
  newSpecPage({
    components: [MgDivider],
    template: () => <mg-divider {...args}></mg-divider>,
  });

describe('mg-divider', () => {
  test.each([undefined, 'regular', 'full'])('size %s', async size => {
    const { root } = await getPage({ size });
    expect(root).toMatchSnapshot();
  });
});
