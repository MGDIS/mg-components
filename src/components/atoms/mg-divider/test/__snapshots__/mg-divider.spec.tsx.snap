// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mg-divider size full 1`] = `
<mg-divider>
  <mock:shadow-root>
    <div class="mg-divider mg-divider--full"></div>
  </mock:shadow-root>
</mg-divider>
`;

exports[`mg-divider size regular 1`] = `
<mg-divider>
  <mock:shadow-root>
    <div class="mg-divider"></div>
  </mock:shadow-root>
</mg-divider>
`;

exports[`mg-divider size undefined 1`] = `
<mg-divider>
  <mock:shadow-root>
    <div class="mg-divider"></div>
  </mock:shadow-root>
</mg-divider>
`;
